power_grid_model_c/power_grid_model_cpp/include/power_grid_model_cpp/buffer.hpp

```diff
@@ -0,0 +1,60 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+#ifndef POWER_GRID_MODEL_CPP_BUFFER_HPP
+#define POWER_GRID_MODEL_CPP_BUFFER_HPP
+
+#include "buffer.h"
+
+#include "basics.hpp"
+#include "handle.hpp"
+
+namespace power_grid_model_cpp {
+class Buffer {
+  public:
+    Buffer(PGM_MetaComponent const* component, Idx size)
+        : component_{component}, size_{size}, buffer_{PGM_create_buffer(handle_.get(), component, size)} {};
+
+    ~Buffer() = default;
```

Comment:
rule of 3-5-0
```suggestion
```

In addition: please add the C++ wrapper to clang tidy (requires creating a target that includes and uses the symbols defined here) and sonar cloud. that should catch these issues early

Reply:
I will probably stick to the rule of 0 in all the classes, but `Model`.

Reply:
I did add in `clang-tidy.yml` the target `power_grid_model_cpp` in the options, but I am unsure if that is enough.

Reply:
nope, a cpp target file is required. In fact, even if a class is defined in a header, it's not checked by clang-tidy unless it's actually used in a function in a cpp file

Reply:
This should then be addressed when the tests are re-routed.

Reply:
Tests added. Resolving.